### ISSUE TITLE
Smithy CLI Support

### ISSUE THREAD

**Issue by iancaffey** [iancaffey/smithy-intellij-plugin#9]
### Background

[Smithy CLI](https://aws.amazon.com/blogs/developer/introducing-the-smithy-cli/) GA'd a few weeks ago. 

Users who wish to migrate off the [Smithy Gradle Plugin](https://smithy.io/2.0/guides/building-models/gradle-plugin.html#smithy-gradle-plugin) onto the CLI will run into dependency resolution issues in IntelliJ.

### Deliverables
- Parse Maven repositories and dependencies from [smithy-build.json](https://smithy.io/2.0/guides/building-models/build-config.html#maven-configuration), register all Maven repositories, add all Maven dependencies to the current Module, and add the Smithy CLI as a Library
- Tool Window which provides all Smithy build actions (similar to Gradle) 
  - Detect the build system being used for Smithy (CLI, Gradle, or "Custom")
    - If using Gradle or some custom/unknown build system, no additional build system support will be provided outside of Maven dependency/repository resolution.
  - Discover the set of commands from Smithy CLI and present them in the UI with an ability to create Run Configurations which invoke the Smithy CLI
  - Process Smithy CLI build output and annotate Smithy files within the editor with the last build execution warnings/errors.

#### References
- [ImportMavenRepositoriesTask](https://github.com/JetBrains/intellij-community/blob/master/plugins/gradle-maven/src/org/jetbrains/plugins/gradle/integrations/maven/ImportMavenRepositoriesTask.java) for how IntelliJ supports this from Gradle build scripts.
- [External Builder API](https://plugins.jetbrains.com/docs/intellij/external-builder-api.html)
- [External System Integration](https://plugins.jetbrains.com/docs/intellij/external-system-integration.html)

**Comment by kubukoz** [comment]
Hi @iancaffey, my team would be able to implement the Maven dependency part of this. If you're not working on it, do you see any issue with it? We'd likely use the LSP implementation as inspiration.

**Comment by iancaffey** [comment]
Hey @kubukoz! No issue at all, implementing the Maven dependency support would be very much appreciated. :)

**Comment by denisrosca** [comment]
Hi @iancaffey ,
I wanted to take a stab at the maven dependency support, but came up a bit short so I'd be nice if I could get some things clarified.
At the moment `smithy-intellij-plugin` doesn't provide any dependency mechanism. To resolve any potential dependencies defined in `smithy-build.json` we would have to either:
- Rely on an existing build system (i.e. if the project is set-up with either maven or gradle) and somehow extend it to be aware of the dependencies specified in the smithy build file.
- Implement our own "build system" which would use the "native" [MavenDependencyResolver](https://github.com/smithy-lang/smithy/blob/main/smithy-cli/src/main/java/software/amazon/smithy/cli/dependencies/MavenDependencyResolver.java).

Does that sound correct?

**Comment by iancaffey** [comment]
Yep, [External System Integration](https://plugins.jetbrains.com/docs/intellij/external-system-integration.html) would likely be the best path forward. 

Since this would be the way to add all of the build features (e.g. auto-import, running build commands, etc.).

**Comment by rezar1** [comment]
Do you have any plans for which version to support? Urgent need
